specs/Deployment/MSIXPackages.md

```diff
@@ -0,0 +1,217 @@
+
+1. Framework
+2. Main
+3. Singleton
+4. Dynamic Dependency Lifetime Manager (DDLM)
+
+## 2.1. Roles
+
+### 2.1.1. Role - Framework
+
+The Framework package is the API delivery vehicle for Windows App SDK. The vast majority of code in
+the project is found here.
+
+Framework packages support concurrent behavior. A user may have multiple versions of a framework
+package in use across processes (process1 using 1.0.1.0, process2 using 1.0.2.0, ...). Updates can
+be registered for a user without disrupting running apps. Windows points running apps to updated
+Framework packages non-disruptively (e.g. when an app's not running), colaescing apps to newer
+versions over time (not necessarily instaneously). This provides for a very flexible servicing
+model.
+
+### 2.1.2. Role - Main
+
+The Main package supplements the Framework package, for when functionality is needed but can't be
+delivered via the Framework package. Mechanisms include [Packaged COM](https://blogs.windows.com/windowsdeveloper/2017/04/13/com-server-ole-document-support-desktop-bridge/),
+[app services](https://docs.microsoft.com/windows/uwp/launch-resume/how-to-create-and-consume-an-app-service),
+[app extensions](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.AppExtensions.AppExtension),
+[background tasks](https://docs.microsoft.com/windows/uwp/launch-resume/create-and-register-a-background-task),
+[startup tasks](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.StartupTask) executing
+on user login, [ApplicationData](https://docs.microsoft.com/uwp/api/Windows.Storage.ApplicationData)
+for data storage scoped to Windows App SDK (not any particular app using Windows App SDK), and
+more.
+
+The primary scenarios include:
+
+* Brokered Access -- processes running in an [AppContainer](https://docs.microsoft.com/windows/win32/secauthz/appcontainer-isolation) requiring
+                     access to resources that cannot be access from within the AppContainer can use
+                     a 'helper process' running at MediumIL and doing work on the API's behalf.
+                     cannot do for itself (i.e. brokered access to a resource).
+* Long-Running Process (aka user service) -- functionality requiring a long-running process
+                                             (especially if its lifetime isn't directly coupled to
+                                             an app's lifetime).
+
+Main packages don't support side-by-side behavior for a user. A user can only have 1 (or 0) Main
+packages in a package family registered at a time. Thus if a user has v1.0.0.0 registered when
+1.0.0.1 is registered it's an *update*, ending with the user having v1.0.0.1 registered instead of
+v1.0.0.0. This poses servicing challenges, as updating the Main package requires any running
+processes from the existing Main package to be terminated.
+
+For these reasons developers are strongly encouraged to only rely on the Main package when
```

Comment:
Seems important to note who "developers" means here. I wouldn't expect Contoso to know whether the WinAppSDK features are implemented in the FWK or the Main packages, nor should it impact them other than how they choose to deliver an unpackaged app.